### ISSUE TITLE
InterruptedExceptions are not handled properly

### ISSUE THREAD

**Issue by abollaert** [freedomotic/freedomotic#348]
Please refer to [Dealing with InterruptedException](https://www.ibm.com/developerworks/library/j-jtp05236/) from IBM DeveloperWorks.

The main issue is that when catching an InterruptedException, the interrupt status flag of the interrupted thread has been cleared.

FTA:

> Interruption is a cooperative mechanism. When one thread interrupts another, the interrupted thread does not necessarily stop what it is doing immediately. Instead, interruption is a way of politely asking another thread to stop what it is doing if it wants to, at its convenience.

> The worst thing you can do with InterruptedException is swallow it -- catch it and neither rethrow it nor reassert the thread's interrupted status. The standard approach to dealing with an exception you didn't plan for -- catch it and log it -- also counts as swallowing the interruption because code higher up on the call stack won't be able to find out about it. (Logging InterruptedException is also just silly because by the time a human reads the log, it is too late to do anything about it.)

In essence, the way you would typically implement a long-running task, is doing something like:

`while (!Thread.currentThread.isInterrupted()) {
// do a bunch of stuff, blocking reads, sleep, whatever
}`

And when you whould want to stop the given task, you can then simply interrupt the thread (in essence, that's what `Future.cancel()` does).

Basically when you have a catch clause for an InterruptedException, instead of logging it, the code should just re-assert the interrupt (`Thread.currentThread().interrupt()`), and optionally log it (not really useful anyway).
